04_basic_struct/basic_struct.c

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: GPL-2.0
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/sysfs.h>
+#include <linux/list.h>
+#include <linux/slab.h>
+
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("Oleksiy Lyubochko <oleksiy.m.lyubochko@globallogic.com>");
+MODULE_DESCRIPTION("A simple example Linux module.");
+MODULE_VERSION("0.01");
+
+#define BUF_SIZE 4096
+#define MODULE_NAME "basic_struct"
+#define STAMP_STRING "List entry:\n'%s'\n"
+
+struct my_data {
+	struct list_head list;
+	char *str;
+};
+
+static LIST_HEAD(my_list_head);
+
+static ssize_t list_show(struct kobject *kobj, struct kobj_attribute *attr,
+			char *buf)
+{
+	struct my_data *ptr;
+	size_t size = 0;
+
+	list_for_each_entry(ptr, &my_list_head, list) {
+		size += snprintf(buf + size, BUF_SIZE, STAMP_STRING, ptr->str);
```

Comment:
When calling snprintf several times, need pass size of free space in buffer: BUF_SIZE - size.